ui-admin/src/forms/designer/questions/ChoicesList.tsx

```diff
@@ -0,0 +1,158 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faChevronDown, faChevronUp, faPlus, faTimes } from '@fortawesome/free-solid-svg-icons'
+import React, { useId } from 'react'
+
+import { CheckboxQuestion, DropdownQuestion, RadiogroupQuestion } from '@juniper/ui-core'
+
+import { Button, IconButton } from 'components/forms/Button'
+import { TextInput } from 'components/forms/TextInput'
+
+type QuestionWithChoices = CheckboxQuestion | DropdownQuestion | RadiogroupQuestion
+
+type ChoicesListProps = {
+  question: QuestionWithChoices
+  readOnly: boolean
+  onChange: (newValue: QuestionWithChoices) => void
+}
+
+/** UI for editing the list of choices for a question. */
+export const ChoicesList = (props: ChoicesListProps) => {
+  const { question, readOnly, onChange } = props
+
+  const labelId = useId()
+
+  return (
+    <div className="mb-3">
+      <p className="mb-2" id={labelId}>Choices</p>
+      <ol aria-labelledby={labelId} className="list-group list-group-numbered mb-1">
+        {question.choices.map((choice, i) => {
+          return (
+            <li
+              key={i}
+              className="list-group-item d-flex"
+            >
+              <div
+                className="flex-grow-1 ms-3"
+                style={{
+                  display: 'grid',
+                  grid: 'auto-flow / max-content auto',
+                  gridColumnGap: '0.5rem',
+                  gridRowGap: '1rem',
+                  alignItems: 'center',
+                  height: 'fit-content'
+                }}
+              >
+                <TextInput
+                  className="form-control"
+                  disabled={readOnly}
+                  label="Text"
+                  labelClassname="mb-0"
+                  value={choice.text}
+                  onChange={value => {
+                    onChange({
+                      ...question,
+                      choices: [
+                        ...question.choices.slice(0, i),
+                        { ...question.choices[i], text: value },
+                        ...question.choices.slice(i + 1)
+                      ]
+                    })
+                  }}
+                />
+                <TextInput
+                  disabled={readOnly}
+                  label="Value"
+                  labelClassname="mb-0"
+                  value={choice.value}
+                  onChange={value => {
+                    onChange({
+                      ...question,
+                      choices: [
+                        ...question.choices.slice(0, i),
+                        { ...question.choices[i], value },
+                        ...question.choices.slice(i + 1)
```

Comment:
same comment as above, seems like a named `changeChoiceValue(index, value)` function would make this more readable